apps/imports/serializers.py

```diff
@@ -0,0 +1,84 @@
+"""
+Copyright 2019 ShipChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from django.conf import settings
+
+from enumfields.drf import EnumSupportSerializerMixin
+from rest_framework_json_api import serializers
+from rest_framework import status
+
+from apps.utils import UpperEnumField, S3PreSignedMixin, UploadStatus
+from .models import ShipmentImport, ProcessingStatus, ShipmentUploadFileType
+
+
+class AbstractSerializer(S3PreSignedMixin, EnumSupportSerializerMixin, serializers.ModelSerializer):
```

Comment:
I think you can refactor these 3 serializers in to two; a base `ShipmentImportSerializer` and then a `ShipmentImportCreateSerializer` that overrides whatever behavior it needs to for creation-specific checks, similar to how the `ShipmentSerializer` is structured.